app/src/main/java/com/quran/labs/androidquran/ui/fragment/QuranAdvancedSettingsFragment.java

```diff
@@ -0,0 +1,462 @@
+package com.quran.labs.androidquran.ui.fragment;
+
+import android.app.Activity;
+import android.app.ProgressDialog;
+import android.content.Context;
+import android.content.DialogInterface;
+import android.content.Intent;
+import android.content.SharedPreferences;
+import android.content.pm.ResolveInfo;
+import android.net.Uri;
+import android.os.AsyncTask;
+import android.os.Build;
+import android.os.Bundle;
+import android.os.Environment;
+import android.preference.Preference;
+import android.preference.PreferenceCategory;
+import android.preference.PreferenceFragment;
+import android.support.v7.app.AlertDialog;
+import android.text.TextUtils;
+import android.widget.Toast;
+
+import com.crashlytics.android.answers.Answers;
+import com.crashlytics.android.answers.CustomEvent;
+import com.quran.labs.androidquran.AdvancedQuranPreferenceActivity;
+import com.quran.labs.androidquran.BuildConfig;
+import com.quran.labs.androidquran.QuranApplication;
+import com.quran.labs.androidquran.QuranImportActivity;
+import com.quran.labs.androidquran.R;
+import com.quran.labs.androidquran.data.Constants;
+import com.quran.labs.androidquran.model.bookmark.BookmarkImportExportModel;
+import com.quran.labs.androidquran.service.util.PermissionUtil;
+import com.quran.labs.androidquran.ui.preference.DataListPreference;
+import com.quran.labs.androidquran.util.QuranFileUtils;
+import com.quran.labs.androidquran.util.QuranSettings;
+import com.quran.labs.androidquran.util.QuranUtils;
+import com.quran.labs.androidquran.util.RecordingLogTree;
+import com.quran.labs.androidquran.util.StorageUtils;
+
+import java.io.File;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+
+import javax.inject.Inject;
+
+import rx.Observable;
+import rx.Subscriber;
+import rx.Subscription;
+import rx.android.schedulers.AndroidSchedulers;
+import rx.functions.Action1;
+import rx.functions.Func1;
+import rx.schedulers.Schedulers;
+import timber.log.Timber;
+
+public class QuranAdvancedSettingsFragment extends PreferenceFragment implements
```

Comment:
let's keep it as 2 and clean it up later since this is mostly a straight forward move of code - perhaps in another PR if it makes sense?